src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs

```diff
@@ -155,6 +155,51 @@ static bool Process(RegexNode node, ref ValueStringBuilder vsb)
             }
         }
 
+        /// <summary>Finds a string literal that is guaranteed to exist in any successful match.</summary>
+        /// <remarks>The exact position of the literal is unknown.</remarks>
+        /// <param name="node">The RegexNode tree node.</param>
+        /// <returns>The found literal, or none if there isn't one.</returns>
+        public static string? FindGuaranteedLiteral(RegexNode node)
```

Comment:
how does all this work when there's more than one BOL in the pattern? eg `^foo\n^baar` ? it's not clear to me how we wouldn't end up returning "baar" here.

Reply:
I can't find many tests with multiple BOL's and literal strings between. I wonder whether it's worth augmenting with a few more.

Reply:
Good catch.